### ISSUE TITLE
A lot of  seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem) cause gemini to failed

### ISSUE THREAD

**Issue by aleksbykov** [scylladb/scylla#10210]
*Installation details*
Scylla version (or git commit hash): `4.4.9-0.20220313.c37f5938f with build-id 94b1a0017f259c65b8478097f52311088a991c57`
Cluster size: 3 nodes (i3.large)
OS (RHEL/CentOS/Ubuntu/AWS AMI): `ami-0e843eda884765abc` (aws: eu-north-1)

Gemini command:
`/$HOME/gemini -d --duration 3h --warmup 30m -c 100 -m mixed -f --non-interactive --cql-features normal --max-mutation-retries 5 --max-mutation-retries-backoff 500ms --async-objects-stabilization-attempts 5 --async-objects-stabilization-backoff 500ms --replication-strategy "{'class': 'SimpleStrategy', 'replication_factor': '3'}" --oracle-replication-strategy "{'class': 'SimpleStrategy', 'replication_factor': '1'}" --test-cluster=10.0.1.156,10.0.3.228,10.0.1.12 --outfile /home/centos/gemini_result_d2c0919f-d2a8-4a89-bb15-7bedbff0f3b3.log --seed 98 --oracle-cluster=10.0.0.28 `
Gemini version: `1.7.5`

Test: `gemini-3h-with-nemesis-test`
Test name: `gemini_test.GeminiTest.test_load_random_with_nemesis`
Test config file(s):
- [gemini-3h-with-nemesis.yaml]([https://github.com/scylladb/scylla-cluster-tests/blob/0d21beeb95740f69f66ef07af45df7d24a22b9d0/test-cases/gemini/gemini-3h-with-nemesis.yaml](https://www.google.com/url?q=https://github.com/scylladb/scylla-cluster-tests/blob/0d21beeb95740f69f66ef07af45df7d24a22b9d0/test-cases/gemini/gemini-3h-with-nemesis.yaml&source=gmail-html&ust=1647339550108000&usg=AOvVaw3XzcUV0R31pLrf2NaJJdxy))

**Issue description**

====================================

During warmup period gemini run only write operations and verify number fo rows with select. Gemini used next schema:
```
Schema: {
    "keyspace": {
        "name": "ks1",
        "replication": {
            "class": "SimpleStrategy",
            "replication_factor": "3"
        },
        "oracle_replication": {
            "class": "SimpleStrategy",
            "replication_factor": "1"
        }
    },
    "tables": [
        {
            "name": "table1",
            "partition_keys": [
                {
                    "name": "pk0",
                    "type": "int"
                },
                {
                    "name": "pk1",
                    "type": "smallint"
                },
                {
                    "name": "pk2",
                    "type": "varint"
                }
            ],
            "clustering_keys": [
                {
                    "name": "ck0",
                    "type": "varchar"
                },
                {
                    "name": "ck1",
                    "type": "smallint"
                },
                {
                    "name": "ck2",
                    "type": "ascii"
                }
            ],
            "columns": [
                {
                    "name": "col0",
                    "type": "date"
                },
                {
                    "name": "col1",
                    "type": "blob"
                },
                {
                    "name": "col2",
                    "type": {
                        "types": {
                            "udt_1134415108_0": "inet",
                            "udt_1134415108_1": "bigint",
                            "udt_1134415108_2": "inet",
                            "udt_1134415108_3": "smallint",
                            "udt_1134415108_4": "ascii",
                            "udt_1134415108_5": "inet",
                            "udt_1134415108_6": "int",
                            "udt_1134415108_7": "decimal"
                        },
                        "type_name": "udt_1134415108",
                        "frozen": true
                    }
                },
                {
                    "name": "col3",
                    "type": {
                        "key_type": "float",
                        "value_type": "int",
                        "frozen": true
                    }
                },
                {
                    "name": "col4",
                    "type": {
                        "key_type": "inet",
                        "value_type": "inet",
                        "frozen": false
                    }
                },
                {
                    "name": "col5",
                    "type": {
                        "types": [
                            "smallint",
                            "decimal",
                            "duration",
                            "boolean",
                            "float",
                            "smallint",
                            "duration",
                            "tinyint",
                            "varint",
                            "uuid",
                            "inet",
                            "ascii"
                        ],
                        "frozen": true
                    }
                },
                {
                    "name": "col6",
                    "type": "tinyint"
                },
                {
                    "name": "col7",
                    "type": "varchar"
                },
                {
                    "name": "col8",
                    "type": "inet"
                },
                {
                    "name": "col9",
                    "type": "varchar"
                }
            ],
            "indexes": [
                {
                    "name": "table1_col6_idx",
                    "column": "col6",
                    "column_idx": 6
                }
            ],
            "materialized_views": [
                {
                    "name": "table1_mv_0",
                    "partition_keys": [
                        {
                            "name": "col1",
                            "type": "blob"
                        },
                        {
                            "name": "pk0",
                            "type": "int"
                        },
                        {
                            "name": "pk1",
                            "type": "smallint"
                        },
                        {
                            "name": "pk2",
                            "type": "varint"
                        }
                    ],
                    "clustering_keys": [
                        {
                            "name": "ck0",
                            "type": "varchar"
                        },
                        {
                            "name": "ck1",
                            "type": "smallint"
                        },
                        {
                            "name": "ck2",
                            "type": "ascii"
                        }
                    ]
                }
            ],
            "known_issues": {
                "https://github.com/scylladb/scylla/issues/3708": true
            }
        }
    ]
}

```

scylla nodes reported a load of next errors:
### node1
```

2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | scylla: [shard 0] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:#012Permits with state active#012memory#011count#011name#012143K#0118#011ks1.table1_col6_idx_index:data-query#0122674K#011139#011ks1.table1:data-query#012#0122817K#011147#011total#012#012Permits with state waiting#012memory#011count#011name#01276K#0111140#011ks1.table1:data-query#012#01276K#0111140#011total#012#012Total: permits: 1287, memory: 2892K
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | sshd[14596]: Accepted publickey for centos from 10.0.3.90 port 56476 ssh2: RSA SHA256:1YmYWh7roz4Y3FSY7N8CetmQNlK/K5FeBPITC8KF0u8
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:39+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)

2022-03-13T17:09:03+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | scylla: [shard 0] compaction - [Compact ks1.table1_mv_0 e9fc1ce0-a2ef-11ec-a078-000000000000] Compacted 2 sstables to [/var/lib/scylla/data/ks1/table1_mv_0-ff703240a2eb11eca078000000000000/md-60-big-Data.db:level=0, ]. 1GB to 1GB (~99% of original) in 159099ms = 7MB/s. ~52608 total partitions merged to 52558.
2022-03-13T17:09:03+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | scylla: [shard 0] compaction - [Compact ks1.table1_col6_idx_index 48d25f40-a2f0-11ec-a078-000000000000] Compacting [/var/lib/scylla/data/ks1/table1_col6_idx_index-febc99b1a2eb11ec9f0f000000000000/md-14-big-Data.db:level=0, /var/lib/scylla/data/ks1/table1_col6_idx_index-febc99b1a2eb11ec9f0f000000000000/md-18-big-Data.db:level=0, ]
2022-03-13T17:09:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | scylla: [shard 0] compaction - [Compact ks1.table1_col6_idx_index 3ca43d10-a2f0-11ec-a078-000000000000] Compacted 2 sstables to [/var/lib/scylla/data/ks1/table1_col6_idx_index-febc99b1a2eb11ec9f0f000000000000/md-22-big-Data.db:level=0, ]. 128MB to 128MB (~99% of original) in 21707ms = 5MB/s. ~256 total partitions merged to 118.
2022-03-13T17:09:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | scylla: [shard 0] compaction - [Compact ks1.table1_col6_idx_index 4995af40-a2f0-11ec-a078-000000000000] Compacting [/var/lib/scylla/data/ks1/table1_col6_idx_index-febc99b1a2eb11ec9f0f000000000000/md-22-big-Data.db:level=0, /var/lib/scylla/data/ks1/table1_col6_idx_index-febc99b1a2eb11ec9f0f000000000000/md-12-big-Data.db:level=0, ]
2022-03-13T17:09:06+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:06+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:06+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:06+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !INFO    | sshd[14708]: Accepted publickey for centos from 10.0.3.90 port 56556 ssh2: RSA SHA256:1YmYWh7roz4Y3FSY7N8CetmQNlK/K5FeBPITC8KF0u8
2022-03-13T17:09:06+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-1 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.156, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
```

### node2
```
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.156#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.156#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 0] storage_proxy - exception during mutation write to 10.0.3.228: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.156#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 0] storage_proxy - exception during mutation write to 10.0.3.228: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.156#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.1.12#0: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 0] storage_proxy - exception during mutation write to 10.0.3.228: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)
2022-03-13T16:39:05+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 0] storage_proxy - exception during mutation write to 10.0.3.228: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 1->0 semaphore)

2022-03-13T17:08:37+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !INFO    | scylla: [shard 0] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:#012Permits with state active#012memory#011count#011name#012297K#01121#011ks1.table1_col6_idx_index:data-query#0123117K#011107#011ks1.table1:data-query#012#0123414K#011128#011total#012#012Permits with state waiting#012memory#011count#011name#0123K#0114271#011ks1.table1:data-query#012#0123K#0114271#011total#012#012Total: permits: 4399, memory: 3417K
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:08:38+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-2 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.3.228, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)

```

### node3
```
2022-03-13T17:08:58+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !INFO    | scylla: [shard 0] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:#012Permits with state active#012memory#011count#011name#012631M#01169#011ks1.table1_mv_0:data-query#012#012631M#01169#011total#012#012Permits with state waiting#012memory#011count#011name#0120B#0118#011ks1.table1:data-query#0120B#01138#011ks1.table1_mv_0:data-query#01214K#0112#011ks1.table1_col6_idx_index:data-query#012#01214K#01148#011total#012#012Total: permits: 117, memory: 631M
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !INFO    | scylla: [shard 1] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:#012Permits with state active#012memory#011count#011name#012625M#01188#011ks1.table1_mv_0:data-query#012#012625M#01188#011total#012#012Permits with state waiting#012memory#011count#011name#0120B#0114#011ks1.table1:data-query#0120B#01111#011ks1.table1_mv_0:data-query#012#0120B#01115#011total#012#012Total: permits: 103, memory: 625M
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:00+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:01+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !INFO    | sshd[14360]: Accepted publickey for centos from 10.0.3.90 port 34180 ssh2: RSA SHA256:1YmYWh7roz4Y3FSY7N8CetmQNlK/K5FeBPITC8KF0u8

2022-03-13T17:09:06+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:07+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !INFO    | sshd[14386]: pam_unix(sshd:session): session closed for user centos
2022-03-13T17:09:07+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !INFO    | systemd-logind: Removed session 651.
2022-03-13T17:09:07+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:07+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:08+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 0] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
2022-03-13T17:09:10+00:00  gemini-with-nemesis-3h-normal-4-4-db-node-436462ad-3 !ERR     | scylla: [shard 1] storage_proxy - Exception when communicating with 10.0.1.12, to read from ks1.table1: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
```

Gemini reported next errors and failed:
```
{"L":"INFO","T":"2022-03-13T16:38:23.731Z","N":"generator","M":"starting partition key generation loop"}
{"L":"INFO","T":"2022-03-13T17:08:53.592Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
{"L":"INFO","T":"2022-03-13T17:08:53.623Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
{"L":"INFO","T":"2022-03-13T17:08:53.677Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
{"L":"INFO","T":"2022-03-13T17:08:54.054Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
{"L":"INFO","T":"2022-03-13T17:08:54.358Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
{"L":"INFO","T":"2022-03-13T17:08:55.390Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
{"L":"INFO","T":"2022-03-13T17:08:55.510Z","N":"validation_job","M":"validation failed for possible async operation","trying_again_in":"500ms"}
Errors detected: []main.JobError{main.JobError{Timestamp:time.Time{wall:0xc083a72af368e42f, ext:1842386589594, loc:(*time.Location)(0x105baa0)}, Message:"Validation failed: row count differ (test has 1 rows, oracle has 0 rows, test is missing rows: [], oracle is missing rows: [pk0=2139040049, \tpk1=-16302, \tpk2=4609149753008074980, \tck0=1KzzN9gJbLDjVtsuNt8VN3rJQv7P9O2Y8/GcouLogY88DjvZz84urAg+3gLSmQc7UB3dY84RfI1sAotWAANvukrgkrecuCgtyYUeyPgyx5OlyWiUNt5FYweGgeFsuRw4HpBGtb82YnEbGBstkh3bVudPmu3WR10/AruzQ8IUc87KQoah9VFTVX9VOzw6yhOANFTYzmxhkShy8rOvozT7BkkiHjRKm/Q0KHULT22wZ1ktlrCoVTGgSa0R0wnVINyXat21I5AJtujRUwy7AVWw1c+Lp1Tf5Z/U39U4KIr+uCyPklkRBLutaUZ1AumK5/UQ2gb2iPMYmtaBZm0ts2yk7LvC8lp4UOZSgAxpj2dESkyAIL7Y6+axzv0C34ddE3yuu8v/8VdBal7RvDidRtV+bLxGUXfQwsunIOGFR+yRfBDehxIUyPzoeLeTKXevac2dgyFEVofIoeTS7hOipGeVTjfi5KM5G4xelZDXEYd6A54amasfLZKJ+fiG4Afdn194QsCWcRvlPzace4E8jL8Wrl0SI/hcqYUdNw573mQxvE4aHVMDOviMIYEW+QPi7lbVaH3Hbdh27SXyy6B81qfjcjca8C299LQEvoDEe6I13ja5YjDYpwIbY8nhDf17oS1VMJaeivqgAb0PCAY3iD+vxeKvGpDLrGcaeBTEG3PYzD5sHeJB/S7e9nZlYsQIY, \tck1=25033, \tck2=UEo0Ut0SQnBepr3HcTE36puuCwDE90/iLTEB9a0m4h60fTYdeKT15oirCynn3sFPflG7FyZxY5ddisaSPb35RqKgaqQVuhW1g56lf0nJRN0i/g+elBTzwYP0ZsYJ2rfmWCNgUspuyHbxyfwWkXVqjwR5gjJwACy1UX47VNSX7kPQgpjO7O4RiFe4/8hA+9dzbZgBGF5Bwlduch6nzDs3YN1O4OscTtXKpc72ni/VQeQaQf2NJbXoR5zovF4i79Ae8L6Uq+lr7fY7DaNANHjJ7PD1Z9sCun2Rm1xj42CKr/9CdIRUMuoM1FtzjJOT8pjmMMY7nuoRxxVjFVJKZSTN6HXZnBIId2AzYo/37vmwFBLinHTuu2J2R3FxDnN5h4Kg])", Query:"SELECT * FROM ks1.table1 WHERE pk0=2139040049 AND pk1=-16302 AND pk2=4609149753008074980 "}}{"L":"WARN","T":"2022-03-13T17:09:00.427Z","N":"sample_results","M":"Errors detected. Exiting."}

```

====================================

Restore Monitor Stack command: `$ hydra investigate show-monitor 436462ad-9118-4529-a40f-0b758a7d17cd`
Show all stored logs command: `$ hydra investigate show-logs 436462ad-9118-4529-a40f-0b758a7d17cd`

Test id: `436462ad-9118-4529-a40f-0b758a7d17cd`

Logs:
grafana - [[https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-gemini-3h-with-nemesis-test-scylla-per-server-metrics-nemesis-20220313_174421-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-gemini-3h-with-nemesis-test-scylla-per-server-metrics-nemesis-20220313_174421-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-gemini-3h-with-nemesis-test-scylla-per-server-metrics-nemesis-20220313_174421-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png%255D(https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-gemini-3h-with-nemesis-test-scylla-per-server-metrics-nemesis-20220313_174421-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png)&source=gmail-html&ust=1647339550108000&usg=AOvVaw21t495bwrO3hQz3uQjIDrk)
grafana - [[https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-overview-20220313_174036-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-overview-20220313_174036-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-overview-20220313_174036-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png%255D(https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174036/grafana-screenshot-overview-20220313_174036-gemini-with-nemesis-3h-normal-4-4-monitor-node-436462ad-1.png)&source=gmail-html&ust=1647339550108000&usg=AOvVaw0PGpbCCQu5yMnv-BmglNlO)
db-cluster - [[https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/db-cluster-436462ad.zip](https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/db-cluster-436462ad.zip)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/db-cluster-436462ad.zip%255D(https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/db-cluster-436462ad.zip)&source=gmail-html&ust=1647339550108000&usg=AOvVaw1Qt7iihW23jGfOEDiMlUDx)
loader-set - [[https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/loader-set-436462ad.zip](https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/loader-set-436462ad.zip)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/loader-set-436462ad.zip%255D(https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/loader-set-436462ad.zip)&source=gmail-html&ust=1647339550108000&usg=AOvVaw0HGVL95HtqupKQALsE08DC)
monitor-set - [[https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/monitor-set-436462ad.zip](https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/monitor-set-436462ad.zip)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/monitor-set-436462ad.zip%255D(https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/monitor-set-436462ad.zip)&source=gmail-html&ust=1647339550108000&usg=AOvVaw0XfLPqqdZVku7P5x8L_Z9j)
sct-runner - [[https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/sct-runner-436462ad.zip](https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/sct-runner-436462ad.zip)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/sct-runner-436462ad.zip%255D(https://cloudius-jenkins-test.s3.amazonaws.com/436462ad-9118-4529-a40f-0b758a7d17cd/20220313_174911/sct-runner-436462ad.zip)&source=gmail-html&ust=1647339550108000&usg=AOvVaw21kvhh-hgf1jY_SI4XKLN3)

[Jenkins job URL]([https://jenkins.scylladb.com/job/scylla-4.4/job/gemini-/job/gemini-3h-with-nemesis-test/99/](https://www.google.com/url?q=https://jenkins.scylladb.com/job/scylla-4.4/job/gemini-/job/gemini-3h-with-nemesis-test/99/&source=gmail-html&ust=1647339550108000&usg=AOvVaw3TktzAL0-Bvpxqgii8Fgqs))

**Comment by aleksbykov** [comment]
Running job with 4.4.8

**Comment by aleksbykov** [comment]
@slivne it looks not a regression for 4.4.9. the issue reproduced with same seed  on 4.4.7, 4.4.8

**Comment by yarongilor** [comment]
Might be reproduced in the same test (after gemini warmup phase):
Got error of:
```
node=gemini-basic-3h-2022-2-oracle-db-node-31b47e0f-1
2022-10-03T11:07:55+00:00 gemini-basic-3h-2022-2-oracle-db-node-31b47e0f-1 !ERR | scylla: [shard 1] storage_proxy - Exception when communicating with 10.4.3.175, to read from ks1.table1_mv_0: seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)
```
then Gemini failed with:
```
Errors detected: []main.JobError{main.JobError{Timestamp:time.Date(2022, time.October, 3, 11, 7, 50, 19939979, time.Local), Message:"Validation failed: unable to load check data from the test store: system failed: Operation timed out for ks1.table1_mv_0 - received only 1 responses from 2 CL=QUORUM."
```


## Installation details

Kernel Version: 5.15.0-1019-aws
Scylla version (or git commit hash): `2022.2.0~rc2-20220919.75d087a2b75a` with build-id `463f1a57b82041a6c6b6441f0cbc26c8ad93091e`
Relocatable Package: http://downloads.scylladb.com/downloads/scylla-enterprise/relocatable/scylladb-2022.2/scylla-enterprise-x86_64-package-2022.2.0-rc2.0.20220919.75d087a2b75a.tar.gz
Cluster size: 3 nodes (i3.large)

Scylla Nodes used in this run:
- gemini-basic-3h-2022-2-oracle-db-node-31b47e0f-1 (34.245.97.42 | 10.4.3.175) (shards: 2)
- gemini-basic-3h-2022-2-db-node-31b47e0f-3 (34.241.252.151 | 10.4.2.72) (shards: 2)
- gemini-basic-3h-2022-2-db-node-31b47e0f-2 (3.250.68.69 | 10.4.2.66) (shards: 2)
- gemini-basic-3h-2022-2-db-node-31b47e0f-1 (52.214.62.157 | 10.4.1.6) (shards: 2)


OS / Image: `ami-00bd31f22bcf5ae1a` (aws: eu-west-1)

Test: `gemini-3h-nondisruptive-nemesis-test`
Test id: `31b47e0f-40ff-4684-af3a-0e8e0d2a6e95`
Test name: `enterprise-2022.2/gemini-/gemini-3h-nondisruptive-nemesis-test`
Test config file(s):

- [gemini-3h-with-nondisruptive-nemesis.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/9818d76f6d1584c8d4652429a6aa1c616b6bb3c2/test-cases/gemini/gemini-3h-with-nondisruptive-nemesis.yaml)

## Issue description

**>>>>>>>**
**Your description here...**
**<<<<<<<**

- Restore Monitor Stack command: `$ hydra investigate show-monitor 31b47e0f-40ff-4684-af3a-0e8e0d2a6e95`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=31b47e0f-40ff-4684-af3a-0e8e0d2a6e95)
- Show all stored logs command: `$ hydra investigate show-logs 31b47e0f-40ff-4684-af3a-0e8e0d2a6e95`

## Logs:
- **db-cluster-31b47e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/db-cluster-31b47e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/db-cluster-31b47e0f.tar.gz)
- **monitor-set-31b47e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/monitor-set-31b47e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/monitor-set-31b47e0f.tar.gz)
- **loader-set-31b47e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/loader-set-31b47e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/loader-set-31b47e0f.tar.gz)
- **sct-runner-31b47e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/sct-runner-31b47e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/31b47e0f-40ff-4684-af3a-0e8e0d2a6e95/20221003_113743/sct-runner-31b47e0f.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2022.2/job/gemini-/job/gemini-3h-nondisruptive-nemesis-test/2/)

**Comment by fgelcer** [comment]
it was seen some time ago on 4.4, but it was never assigned, so i'm setting bot master and release triage, so we can address it

**Comment by wmitros** [comment]
This looks like another issue fixed by https://github.com/scylladb/scylladb/pull/20887.
Is there a corresponding test for some up-to-date Scylla version? If so we can run it. 
Otherwise, even though there's a suspicious spike in `Writes blocked on commitlog` in metrics, I think we can close this regardless

**Comment by kostja** [comment]
@aleksbykov could you please try re-running this against the latest master? I disagree with @wmitros, the logs are incomplete, so he's right that what we see is inconclusive, but I think the root cause is the same and the problem should be fixed by  #20887

**Comment by aleksbykov** [comment]
Started job with reproduce for latest master: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/gemini-repo/job/gemini-reproduce-1794-lsa-grow/

**Comment by aleksbykov** [comment]
@kostja , I reran job several times. On latest run only get similar reproducing:
```
2024-11-23T11:40:39.402+00:00 gemini-with-nemesis-3h-normal-maste-db-node-03d95603-1     !INFO | scylla[7022]:  [shard 0:stmt] reader_concurrency_semaphore - (rate limiting dropped 214 similar messages) Semaphore user with 2/100 count and 40609/174986362 memory resources: timed out, dumping permit diagnostics:
Trigger permit: count=0, memory=0, table=ks1.table1_mv_0, operation=data-query, state=waiting_for_admission
Identified bottleneck(s): CPU

permits count   memory  table/operation/state
1       1       24K     ks1.table1_mv_0/data-query/active/await
1       1       16K     ks1.table1/data-query/active/need_cpu
771     0       0B      ks1.table1_mv_0/data-query/waiting_for_admission
4       0       0B      ks1.table1_col6_idx_index/data-query/waiting_for_admission
5677    0       0B      ks1.table1/data-query/waiting_for_admission

6454    2       40K     total

Stats:
permit_based_evictions: 0
time_based_evictions: 0
inactive_reads: 0
total_successful_reads: 46493753
total_failed_reads: 721
total_reads_shed_due_to_overload: 0
total_reads_killed_due_to_kill_limit: 0
reads_admitted: 46493784
reads_enqueued_for_admission: 42784176
reads_enqueued_for_memory: 0
reads_admitted_immediately: 3716781
reads_queued_because_ready_list: 28877795
reads_queued_because_need_cpu_permits: 13906381
reads_queued_because_memory_resources: 0
reads_queued_because_count_resources: 0
reads_queued_with_eviction: 0
total_permits: 46500957
current_permits: 6454
need_cpu_permits: 2
awaits_permits: 1
disk_reads: 2
sstables_read: 3

```


## Packages

Scylla version: `6.3.0~dev-20241122.e2e6f4f441be` with build-id `2493a7aae1f855d3df502197f757822b6afc1033`





Kernel Version: `6.8.0-1019-aws`


## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

*Describe your issue in detail and steps it took to produce it.*

## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Cluster size: 3 nodes (i4i.2xlarge)

Scylla Nodes used in this run:
- gemini-with-nemesis-3h-normal-maste-db-node-03d95603-2 (18.215.188.59 | 10.12.2.20) (shards: 7)
- gemini-with-nemesis-3h-normal-maste-db-node-03d95603-3 (44.198.167.208 | 10.12.2.113) (shards: 7)
- gemini-with-nemesis-3h-normal-maste-oracle-db-node-03d95603-1 (3.236.129.6 | 10.12.0.80) (shards: 30)
- gemini-with-nemesis-3h-normal-maste-db-node-03d95603-4 (98.80.177.8 | 10.12.0.72) (shards: 7)
- gemini-with-nemesis-3h-normal-maste-db-node-03d95603-1 (3.238.62.185 | 10.12.0.16) (shards: 7)


OS / Image: `ami-0c8445a8fe236d71f` (aws: undefined_region)

Test: `gemini-3h-with-nemesis-test`
Test id: `03d95603-6305-4626-921e-b318860baf68`
Test name: `scylla-staging/abykov/gemini-3h-with-nemesis-test`
Test method: `gemini_test.GeminiTest.test_load_random_with_nemesis`
Test config file(s):

- [gemini-3h-with-nemesis.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/50020224662a89759386ce03310af152e30c5eaa/test-cases/gemini/gemini-3h-with-nemesis.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 03d95603-6305-4626-921e-b318860baf68`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=03d95603-6305-4626-921e-b318860baf68)
- Show all stored logs command: `$ hydra investigate show-logs 03d95603-6305-4626-921e-b318860baf68`


## Logs:
- **gemini-with-nemesis-3h-normal-maste-db-node-03d95603-2** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_094338/gemini-with-nemesis-3h-normal-maste-db-node-03d95603-2-03d95603.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_094338/gemini-with-nemesis-3h-normal-maste-db-node-03d95603-2-03d95603.tar.gz)
- **db-cluster-03d95603.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/db-cluster-03d95603.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/db-cluster-03d95603.tar.gz)
- **sct-runner-events-03d95603.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/sct-runner-events-03d95603.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/sct-runner-events-03d95603.tar.gz)
- **sct-03d95603.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/sct-03d95603.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/sct-03d95603.log.tar.gz)
- **loader-set-03d95603.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/loader-set-03d95603.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/loader-set-03d95603.tar.gz)
- **monitor-set-03d95603.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/monitor-set-03d95603.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/monitor-set-03d95603.tar.gz)
- **parallel-timelines-report-03d95603.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/parallel-timelines-report-03d95603.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/03d95603-6305-4626-921e-b318860baf68/20241123_143444/parallel-timelines-report-03d95603.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-staging/job/abykov/job/gemini-3h-with-nemesis-test/3/)
[Argus](https://argus.scylladb.com/test/461a17df-db14-4b5d-a78b-34c0374e7bc6/runs?additionalRuns[]=03d95603-6305-4626-921e-b318860baf68)
</details>
